tests/core/test_dag_data.py

```diff
@@ -169,50 +101,54 @@ def test_edge_index_kth_order_dag():
     e3 = DAGData.edge_index_kth_order(edge_index, k=3)
     assert equal(
         e3,
-        IntTensor(
-            [[[0, 1, 2], [1, 2, 3], [2, 3, 4]], [[1, 2, 3], [2, 3, 4], [3, 4, 5]]]
-        ),
+        IntTensor([[[0, 1, 2], [1, 2, 3], [2, 3, 4]], [[1, 2, 3], [2, 3, 4], [3, 4, 5]]]),
     )
 
     e4 = DAGData.edge_index_kth_order(edge_index, k=4)
-    assert equal(
-        e4, IntTensor([[[0, 1, 2, 3], [1, 2, 3, 4]], [[1, 2, 3, 4], [2, 3, 4, 5]]])
-    )
+    assert equal(e4, IntTensor([[[0, 1, 2, 3], [1, 2, 3, 4]], [[1, 2, 3, 4], [2, 3, 4, 5]]]))
 
     e5 = DAGData.edge_index_kth_order(edge_index, k=5)
     assert equal(e5, IntTensor([[[0, 1, 2, 3, 4]], [[1, 2, 3, 4, 5]]]))
 
+    e6 = DAGData.edge_index_kth_order(edge_index, k=6)
+    assert equal(e6, IntTensor([]))
 
-def test_path_mapping():
-    mapping = {0: 1, 1: 1, 2: 0, 3: 0, 4: 2, 5: 2}
 
-    e1 = IntTensor([[0, 1, 2, 3, 4], [1, 2, 3, 4, 5]]).to(config["torch"]["device"])
-    m_e1 = PathData.map_nodes(e1, mapping)
+def test_edge_index_k_weighted(simple_temporal_graph):
+    dag = temporal_graph_to_event_dag(simple_temporal_graph, delta=5, sparsify=True)
+    paths = DAGData.from_temporal_dag(dag)
+
+    e1, w1 = DAGData.edge_index_k_weighted(paths, k=1)
 
     assert equal(
-        m_e1,
-        IntTensor([[1, 1, 0, 0, 2], [1, 0, 0, 2, 2]]).to(config["torch"]["device"]),
-    )
+        e1, IntTensor([[0, 1, 2, 2], [1, 2, 3, 4]]).to(config["torch"]["device"])
+    )  # a -> b | b -> c | c -> d | c -> e
+
+    assert equal(w1, tensor([1.0, 1.0, 1.0, 1.0]).to(config["torch"]["device"]))
+
+    e2, w2 = DAGData.edge_index_k_weighted(paths, k=2)
+    assert equal(
+        e2,
+        IntTensor([[[0, 1], [1, 2], [1, 2]], [[1, 2], [2, 3], [2, 4]]]).to(config["torch"]["device"]),
+    )  # a-b -> b-c | b-c -> c-d | b-c -> c-e
 
-    e2 = IntTensor([[[0, 1], [2, 3]], [[1, 2], [3, 4]]]).to(config["torch"]["device"])
-    m_e2 = PathData.map_nodes(e2, mapping)
+    assert equal(w2, tensor([1.0, 1.0, 1.0]).to(config["torch"]["device"]))
 
+    e3, w3 = DAGData.edge_index_k_weighted(paths, k=3)
     assert equal(
-        m_e2,
-        IntTensor([[[1, 1], [0, 0]], [[1, 0], [0, 2]]]).to(config["torch"]["device"]),
+        e3,
+        IntTensor([[[0, 1, 2], [0, 1, 2]], [[1, 2, 3], [1, 2, 4]]]).to(config["torch"]["device"]),
     )
 
+    assert equal(w3, tensor([1.0, 1.0]).to(config["torch"]["device"]))  # a-b-c -> b-c-d | a-b-c -> b-c-e
 
-def test_lift_order_dag():
-    e1 = tensor([[[0], [1], [1], [3]], [[1], [2], [3], [4]]])
-    x = DAGData.lift_order_dag(e1)
-    assert equal(x, IntTensor([[[0, 1], [0, 1], [1, 3]], [[1, 2], [1, 3], [3, 4]]]))
+    with pytest.raises(ValueError):
```

Comment:
This is why `test_edge_index_k_weighted` currently fails. I.e. if you choose an order that is larger than the length of the longest `DAG` the method returns a RunTimeError. Should this stay this way or should we instead return a more meaningful error message or just an empty tensor?

Reply:
We should return an error. Need to see how and where to catch this ...